lib/src/base/paged_layout_builder.dart

```diff
@@ -260,7 +261,7 @@ class _PagedLayoutBuilderState<PageKeyType, ItemType>
       // This behaviour is okay because we make sure not to excessively request pages.
       final hasPassedTrigger = index >= triggerIndex;
 
-      if (_hasNextPage && hasPassedTrigger) {
+      if (_hasNextPage && hasPassedTrigger && !_state.isLoading) {
```

Comment:
This one i didn't understand why it is needed, why this method is called twice is a mystery.